### PR TITLE
Add version check for 1.3.1, add ansible task

### PR DESCRIPTION
**Pull request by vasukulkarni**
Version check for 1.3.1  bit of restructure for overrides and add ansible task
 
Signed-off-by: Vasu Kulkarni <vasu@redhat.com>

### CODE REVIEW

**Comment by zmc** [comment]
The commit message isn't clear (I can't parse it). Overrides don't seem to be affected by this commit either.

**Comment by vasukulkarni** [comment]
@zmc: commit message should be fixed now

**Comment by dmick** [comment]
This is at least three commits, and each of those separate commits should have messages explaining the problem clearly and why this code resolves the problem

**Comment by vasukulkarni** [comment]
@dmick: if you want 3 different commits and message for each commit, I can force push another.

**Comment by dmick** [comment]
3 commits is definitely better, but please put some real information in the commit messages, like I asked: explaining the problem clearly and why this code resolves the problem.  It only takes you a few tens of seconds, but can save a lot of time in six months when we revisit this

**Comment by vasukulkarni** [comment]
@dmick : I already pushed 3 commits in the morning, Each commit now explains what it does.

**Comment by zmc** [comment]
@vasukulkarni @dmick is asking you for commit messages "explaining the problem clearly and why this code resolves the problem". Your commit messages say what they do, yes, but they don't say what the problem they're solving is. This seems like a reasonable request.

**Comment by vasukulkarni** [comment]
@dmick: I have added the required info in the additional lines of commit message it should be clear now what it solves.

File: teuthology/task/install.py [hunk 1]
```diff
@@ -1175,17 +1176,26 @@ def task(ctx, config):
...
 
     if config.get('rhbuild'):
-        with contextutil.nested(
-            lambda: rh_install(ctx=ctx, config=config),
-            lambda: ship_utilities(ctx=ctx, config=None)
-        ):
-            yield
+        if config.get('playbook'):
+            ansible_config=dict(config)
+            #remove key not required by ansible task
```
Comment (dmick) [inline teuthology/task/install.py]: still no space here?

File: teuthology/task/install.py [hunk 2]
```diff
@@ -1175,17 +1176,26 @@ def task(ctx, config):
...
+    if config.get('rhbuild'):
+        rhbuild = config.get('rhbuild')
+        log.info("Build is %s " % rhbuild)
+
     flavor = get_flavor(config)
     log.info("Using flavor: %s", flavor)
 
     ctx.summary['flavor'] = flavor
+    nested_tasks = [lambda: rh_install(ctx=ctx, config=config),
+                     lambda: ship_utilities(ctx=ctx, config=None)]
```
Comment (dmick) [inline teuthology/task/install.py]: the lambdas should still align?

**Pull request merged**